### ISSUE TITLE
phunt-cli build id not injected by rollup

### ISSUE THREAD

**Issue by jelovac** [112dev/phunt#8]
### Description

The BUILD_ID environment variable is not passed to the Rollup node process which builds and bundles `phunt-cli`. As a result when we run the version command the BUILD_ID will display unknown.

### TODO

- [x] Investigate
- [x] Fix

### Test case

1. Run `BUILD_ID=123 pnpm build`
2. Check dist/phunt-cli.js at the very bottom in the code you will have the version block.
  ```
app.version(`Version: 1.0.0-beta.0
Build Id: unknown
Build Date: Sun, 29 Sep 2024 17:27:16 GMT
License: Apache-2.0`);
  ```

By design if the env variable is set rollup's plugin replace should set BUILD_ID.

### Relevant files

- [apps/phunt-cli/rollup.config.mjs:46](https://github.com/112dev/phunt/blob/0c60276069c2e15055a00c787004f98b0789324c/apps/phunt-cli/rollup.config.mjs#L46)
- [.github/workflows/release.yml#L51](https://github.com/112dev/phunt/blob/0c60276069c2e15055a00c787004f98b0789324c/.github/workflows/release.yml#L51)

**Comment by jelovac** [comment]
Figured it out. We run the build task through turbo(repo). And turbo by default excludes all environment variables which are not explicitly stated or match common JS frameworks patterns.

See: 
- https://turbo.build/repo/docs/reference/configuration#globalenv
- https://turbo.build/repo/docs/reference/configuration#globalpassthroughenv
- https://turbo.build/repo/docs/reference/configuration#env

Currently BUILD_ID = github.run_id 

If we set BUILD_ID as a globalenv it will always cause a cache miss, which will make caching useless.

If we set BUILD_ID as a globalpassthroughenv it will not cause a cache miss. However, it might make a confusion. Because the ID will be the id of some previous job which got cached. Not a good solution either..

Time to sleep on it.
